appinventor/appengine/src/com/google/appinventor/client/Ode.java

```diff
@@ -1754,7 +1756,7 @@ private boolean shouldShowWelcomeDialog() {
     } else {
       uversion = Integer.parseInt(value);
     }
-    if (uversion >= splashConfig.version) {
+    if ((uversion >= splashConfig.version) && splashConfig.version != -2) { // -2 means always show the splash screen!
```

Comment:
Rather than doing this here, it might make more sense for readability to add:

```
if (splashConfig.version == -2) {  // Always show splash if version is -2
  return true;
}
```

after the check against 0 at the top of the function.